src/device-registry/utils/generate-filter.js

```diff
@@ -580,6 +581,12 @@ const generateFilter = {
         filter["name"] = device_id;
       }
 
+      if (device_codes) {
+        let deviceCodesArray = device_codes.split(",");
```

Comment:
Allow user to provide a comma separated list of device codes in the query parameter for` device_codes` when querying for device details.

All matches will be returned.